framework/tlvf/src/include/tlvf/CmduMessage.h

```diff
@@ -19,62 +20,36 @@ namespace ieee1905_1 {
 class CmduMessage {
 
 public:
-    CmduMessage();
+    CmduMessage() = delete;
+    CmduMessage(uint8_t *buff, size_t buff_len);
     ~CmduMessage();
 
 public:
     std::shared_ptr<cCmduHeader> getCmduHeader() const;
-    
-    template <class T, class = typename std::enable_if<std::is_base_of<BaseClass, T>::value>::type>
-    std::shared_ptr<T> addClass()
-    {
-        std::shared_ptr<T> ptr;
-        if (m_cmdu_header) {
-            if (m_class_vector.size() == 0) {
-                ptr = std::make_shared<T>(m_cmdu_header, m_parse, m_swap);
-            } else {
-                ptr = std::make_shared<T>(m_class_vector.back(), m_parse, m_swap);
-            }
-            if (!ptr || ptr->isInitialized() == false) {
-                return std::shared_ptr<T>();
-            } else {
-                m_class_vector.push_back(std::shared_ptr<BaseClass>(ptr));
-            }
-        } else {
-            return std::shared_ptr<T>();
-        }
-        return ptr;
-    }
 
+    ieee1905_1::TlvList tlvs;
+
+    template <class T> std::shared_ptr<T> addClass() { return tlvs.addClass<T>(); }
```

Comment:
Same here: please remove the "forwarding" wrappers and use tlvs directly, or make a more relevant wrapper if appropriate.

Reply:
Thinking about this again I don't get what you mean here. `addClass` is widely used, changing everywhere to `cmdu_rx.tlvs.addClass` is a huge change, and although it can be done I see no reason not to encapsulate that with this forwarding wrapper - can you explain whats wrong with it?

Reply:
I think I put this comment on the wrong line. Or maybe I was just wrong :-). Anyway in the final review it will turn up again if it really was relevant.